sonar-java-plugin/src/test/java/org/sonar/plugins/java/JavaRulesDefinitionTest.java

```diff
@@ -68,6 +68,11 @@ public void test_creation_of_rules() {
     RulesDefinition.Rule unusedPrivateMethodRule = repository.rule("UnusedPrivateMethod");
     assertThat(unusedPrivateMethodRule.activatedByDefault()).isEqualTo(true);
 
+    // rule templates are manually defined
+    assertThat(repository.rules().stream()
```

Comment:
This lacks the opposite testing : if I add a new rule template how will I know I actually properly declared it as a rule template ?

Reply:
I agree. We were not testing template rules before, but I don't know how to put in place one without hardcoding the number of non-template rule that we currently have... Which will requires to edit the test with each new rule. I'm not sure we want that.
Otherwise I will simply go for:
```
assertThat(rules.stream().filter(rule -> !rule.template())).hasSize(445 - 6);
```

Reply:
This is a convuleted test of the number of rules.